save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/vuln/VulnerabilityGeneralInfoProps.kt

```diff
@@ -343,10 +347,54 @@ val vulnerabilityGeneralInfoProps: FC<VulnerabilityGeneralInfoProps> = FC { prop
                         className = ClassName("font-weight-bold text-primary-blue mb-4")
                         +"Credits"
                     }
-                    cosv.credits?.forEach { credit ->
-                        val savePrefix = if (credit.name in saveContributors.map { it.name }) "[SAVE] " else ""
-                        p {
-                            +"$savePrefix${credit.name} <${credit.contact?.joinToString { it }}>"
+
+                    cosv.credits!!.forEachIndexed { ind, credit ->
+                        val userPrefix = if (credit.name in saveContributors.map { it.name }) "[SAVE] user" else "External user"
+                        if (props.isEditDisabled) {
+                            div {
+                                className = ClassName("row ml-1")
+                                pre {
+                                    +"$userPrefix\n${credit.name} ${credit.contact?.joinToString(PRETTY_DELIMITER)}"
+                                }
+                            }
+                        } else {
+                            div {
+                                className = ClassName("row")
+                                inputTextFormRequired {
+                                    form = InputTypes.VULN_CREDIT_NAME
+                                    textValue = credit.name
+                                    classes = "col-6 pl-2 pr-2 mt-3 text-left"
+                                    name = userPrefix
+                                    validInput = credit.name.isNotBlank() && credit.name.isValidName()
+                                    onChangeFun = { event ->
+                                        val updatedCredit = credit.copy(name = event.target.value)
+                                        val updatedCredits = cosv.credits as MutableList<Credit>
+                                        updatedCredits[ind] = updatedCredit
+                                        props.setVulnerability { vulnerability ->
+                                            vulnerability?.copy(
+                                                cosv = cosv.copy(credits = updatedCredits)
+                                            )
+                                        }
+                                    }
+                                }
+                                inputTextFormRequired {
+                                    form = InputTypes.VULN_CREDIT_CONTACTS
+                                    textValue = credit.contact!!.joinToString(PRETTY_DELIMITER)
```

Comment:
I guess we need to allow to set contacts separately instead of a single string with splitting

Reply:
It will require some additional work (dynamically add new inputforms for new contacts and so on), maybe its better to do in next step